File: Models/PracticeTask.cs
using System;
using System.Collections.Generic;

namespace PianistApp.Models;

public sealed class PracticeTask
{
    public string Name { get; init; } = string.Empty;
    public string Description { get; init; } = string.Empty;
    public string FingeringDiagram { get; init; } = string.Empty;
    public IReadOnlyList<FingeredNote> FingeringNotes { get; init; } = Array.Empty<FingeredNote>();
    public IReadOnlyList<int> ExpectedNotes { get; init; } = Array.Empty<int>();
    public bool RequiresMidiInput { get; init; } = true;
    public bool RequireMetronome { get; init; }
    public int TempoBpm { get; init; } = 60;
    public int BeatToleranceMs { get; init; } = 120;
    public double MinNoteAccuracy { get; init; } = 0.95;
    public double MinMetronomeAccuracy { get; init; } = 0.9;
}

File: lib/models/practice_task.dart
import 'fingered_note.dart';

class PracticeTask {
  const PracticeTask({
    required this.id,
    required this.title,
    required this.description,
    required this.expectedNotes,
    required this.metronomeRequired,
    required this.tempoBpm,
  });

  final String id;
  final String title;
  final String description;
  final List<FingeredNote> expectedNotes;
  final bool metronomeRequired;
  final int tempoBpm;
}

class PracticeSection {
  const PracticeSection({
    required this.title,
    required this.tasks,
  });

  final String title;
  final List<PracticeTask> tasks;
}
